python/src/aiconfig/default_parsers/openai.py

```diff
@@ -505,27 +529,48 @@ def add_prompt_as_message(
         if output.output_type == "execute_result":
             assert isinstance(output, ExecuteResult)
             output_data = output.data
-            # TODO (rossdanlm): Support function call handling as output_data
-            if isinstance(output_data, str):
-                if output.metadata["rawResponse"].get("role", "") == "assistant":
-                    output_message = {
-                        "content": output_data,
-                        "role": "assistant",
-                    }
-                    function_call = output.metadata["rawResponse"].get("function_call", None)
-                    tool_calls = output.metadata["rawResponse"].get("tool_calls", None)
-                    if function_call is not None:
-                        output_message["function_call"] = function_call
-                    if tool_calls is not None:
-                        output_message["tool_calls"] = tool_calls
-                    messages.append(output_message)
+            role = output.metadata.get("assistant", None) or \
+                ("rawResponse" in output.metadata and
+                    output.metadata["rawResponse"].get("assistant", None))
```

Comment:
This is incorrect. Can you check if TS or Python implementations have this issue elsewhere in your stack?

```suggestion
            role = output.metadata.get("role", None) or \
                ("rawResponse" in output.metadata and
                    output.metadata["rawResponse"].get("role", None))
```

Reply:
Also perhaps we should default role to "assistant"

Reply:
Yea, just copy pasted wrong my bad, should be `role`.

> Can you check if TS or Python implementations have this issue elsewhere in your stack?

I don't have this issue anywhere else (https://github.com/lastmile-ai/aiconfig/pull/610/files#diff-e044912cdb054134cac74868d76969daa92c6dd7c1238805672e023f9b7aa92d), this was just a typo

> Also perhaps we should default role to "assistant"

We shouldn't be doing this. Role can be one of 5 different types, not just "assistant", so we need to check for that explicitly: https://github.com/openai/openai-node/blob/b595cd953a704dba2aef4c6c3fa431f83f18ccf9/src/resources/chat/completions.ts#L514